chained/water_magnet_downloader.go

```diff
@@ -0,0 +1,52 @@
+package chained
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/anacrolix/torrent"
+)
+
+type magnetDownloader struct {
+	magnetURL string
+}
+
+// NewMagnetDownloader creates a new WASMDownloader instance.
+func NewMagnetDownloader(magnetURL string) WASMDownloader {
+	return &magnetDownloader{
+		magnetURL: magnetURL,
+	}
+}
+
+// DownloadWASM downloads the WASM file from the given URL.
+func (d *magnetDownloader) DownloadWASM(ctx context.Context, w io.Writer) error {
+	cfg := torrent.NewDefaultClientConfig()
+	path, err := os.MkdirTemp("/tmp", "torrent")
+	if err != nil {
+		return fmt.Errorf("failed to create temp dir: %w", err)
+	}
+	cfg.DataDir = path
+	client, err := torrent.NewClient(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to create torrent client: %w", err)
+	}
+	defer client.Close()
+
+	t, err := client.AddMagnet(d.magnetURL)
```

Comment:
Yeah it is more efficient. It will sit idle, so leave the client in `magnetDownloader`. That also means it will actually seed stuff for longer than it takes to download too, which I assume is the intention, but fine if not.